server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/CustomQueryScorer.java

```diff
@@ -65,9 +65,7 @@ protected WeightedSpanTermExtractor newTermExtractor(String defaultField) {
         @Override
         protected void extractUnknownQuery(Query query,
                                            Map<String, WeightedSpanTerm> terms) throws IOException {
-            if (terms.isEmpty()) {
-                extractWeightedTerms(terms, query, 1F);
-            }
+            extractWeightedTerms(terms, query, 1F);
```

Comment:
Can you explain why this has changed?

Reply:
This is tricky, I meant to call it out explicitly! Without this fix, the `plain` highlighter does not work on `combined_field` queries. Specifically, it will only highlight the first term, and fail to highlight subsequent terms because of the `terms.isEmpty()` check.

My understanding is that this was added in #15516 to work around a Lucene issue. The Lucene issue was later fixed: https://issues.apache.org/jira/browse/LUCENE-7112. All the tests added in #15516 still pass with this change.